### PR TITLE
ddl: Unify drop table index and tiny clean up

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: ddl/reorg.go [hunk 1]
```diff
@@ -190,14 +191,10 @@ func (d *ddl) delKeysWithPrefix(prefix kv.Key) error {
 			return nil
 		})
 
-		if err != nil {
+		// if err or delete no keys, returns.
+		if err != nil || len(keys) == 0 {
```
Comment (qiuyesuifeng) [inline ddl/reorg.go]: We should keep check err and value separately, like old code.

Comment (zimulala) [inline ddl/reorg.go]: got it.

**Comment by qiuyesuifeng** [comment]
LGTM 
@siddontang  PTAL

**Comment by zimulala** [comment]
PTAL @siddontang

**Comment by siddontang** [comment]
No need to handle drop column data?

**Comment by zimulala** [comment]
@siddontang 
no, because drop column need `handle` and `columnID` to form the key, can't delete with prefix.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**